lib/markbind/lib/parser.js

```diff
@@ -131,7 +136,8 @@ Parser.prototype._preprocess = function (node, context, config) {
       actualFilePath = calculateBoilerplateFilePath(element.attribs.boilerplate, filePath, config);
       this.boilerplateIncludeSrc.push({ from: context.cwf, to: actualFilePath });
     }
-    if (!utils.fileExists(actualFilePath)) {
+    const isOptional = element.name === 'include' && _.hasIn(element.attribs, 'if-exists');
+    if (!utils.fileExists(actualFilePath) && !isOptional) {
```

Comment:
Since this portion of the entire code is dedicated to handle missing files, the logic for creating an empty node for an optional include can be handled directly here, instead of being handled separately in line 153, 157-160:

```javascript
if (!utils.fileExists(actualFilePath) {
  if (isOptional) {
    return createEmptyNode();
  }
  
  this.missingIncludeSrc.push({ from: context.cwf, to: actualFilePath });
  ...
}
```